components/p3a/BUILD.gn

```diff
@@ -33,19 +34,30 @@ static_library("p3a") {
 
   deps = [
     "//base",
-    "//brave/components/brave_referrals/common",
+    "//brave/components/brave_referrals/buildflags",
     "//brave/components/brave_stats/browser",
     "//brave/components/p3a:buildflags",
     "//brave/components/p3a_utils",
     "//brave/components/version_info",
     "//brave/vendor/brave_base",
     "//components/metrics",
     "//components/prefs",
-    "//content/public/browser",
-    "//content/public/common",
     "//services/network/public/cpp",
     "//url",
   ]
+
+  if (enable_brave_referrals) {
```

Comment:
Is there a reason why checking `enable_brave_referrals` now not before?

Reply:
Previously it was enabled on all platforms building this so they didn't necessarily have to check it before even though they should have